dali/operators/audio/preemphasis_filter_op.cu

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/audio/preemphasis_filter_op.h"
+#include <vector>
+
+namespace dali {
+
+namespace detail {
+
+template <typename OutputType, typename InputType>
+struct SampleDescriptor {
+  const InputType *in;
+  OutputType *out;
+  float coeff;
+  int64_t size;
+};
+
+template <typename OutputType, typename InputType>
+void __global__ PreemphasisFilterKernel(const SampleDescriptor<OutputType, InputType> *samples) {
+  const auto &sample = samples[blockIdx.y];
```

Comment:
How much we can get if we won't bound 1 block to one sample, but have each block process part of a full sample depending on the size (to improve utilization for small samples)?

Reply:
It's not block per sample - it's `gridDim.x` blocks per sample. There will be some imbalance across blocks, but with some luck we'll not end up with one block processing one huge sample when others are done.